tests/tests.rs

```diff
@@ -207,37 +217,48 @@ fn compare_generated_header(
     let (expected, rustfmt_stderr) = rustfmt(expected);
     println!("{}", rustfmt_stderr);
 
-    if actual == expected {
-        if !actual.is_empty() {
-            return Ok(());
-        }
+    if actual.is_empty() {
         return Err(Error::new(
             ErrorKind::Other,
             "Something's gone really wrong!",
         ));
     }
 
-    println!("{}", rustfmt_stderr);
+    if actual != expected {
+        println!("{}", rustfmt_stderr);
 
-    println!("diff expected generated");
-    println!("--- expected: {:?}", looked_at.last().unwrap());
-    println!("+++ generated from: {:?}", header);
+        println!("diff expected generated");
+        println!("--- expected: {:?}", looked_at.last().unwrap());
+        println!("+++ generated from: {:?}", header);
 
-    for diff in diff::lines(&expected, &actual) {
-        match diff {
-            diff::Result::Left(l) => println!("-{}", l),
-            diff::Result::Both(l, _) => println!(" {}", l),
-            diff::Result::Right(r) => println!("+{}", r),
+        for diff in diff::lines(&expected, &actual) {
+            match diff {
+                diff::Result::Left(l) => println!("-{}", l),
+                diff::Result::Both(l, _) => println!(" {}", l),
+                diff::Result::Right(r) => println!("+{}", r),
+            }
+        }
+
+        // Overwrite the expectation with actual output.
+        if env::var_os(OVERWRITE_ENV_VAR).is_some() {
+            let mut expectation_file =
+                fs::File::create(looked_at.last().unwrap())?;
+            expectation_file.write_all(actual.as_bytes())?;
         }
+
+        return Err(Error::new(ErrorKind::Other, "Header and binding differ! Run with BINDGEN_OVERWRITE_EXPECTED=1 in the environment to automatically overwrite the expectation."));
     }
 
-    // Overwrite the expectation with actual output.
-    if env::var_os(OVERWRITE_ENV_VAR).is_some() {
-        let mut expectation_file = fs::File::create(looked_at.last().unwrap())?;
-        expectation_file.write_all(actual.as_bytes())?;
+    if check_roundtrip {
+        let roundtrip_builder = builder_from_flags(flags.into_iter())?.0;
```

Comment:
FYI, there are cases the bindgen program would given you an error but `builder_from_flags()` will not show an error.
Sadly, to properly handle, I think we need to execute a sub-process.

https://github.com/rust-lang/rust-bindgen/pull/1819#discussion_r446752135

Reply:
Right, this _ought_ to work :)